source/common/http/codec_client.cc

```diff
@@ -122,7 +122,12 @@ void CodecClient::onReset(ActiveRequest& request, StreamResetReason reason) {
 void CodecClient::onData(Buffer::Instance& data) {
   bool protocol_error = false;
   try {
-    codec_->dispatch(data);
+    auto status = codec_->dispatch(data);
+    if (!status.ok()) {
+      ENVOY_CONN_LOG(debug, "protocol error: {}", *connection_, status.message());
+      close();
+      protocol_error = true;
+    }
   } catch (CodecProtocolException& e) {
     ENVOY_CONN_LOG(debug, "protocol error: {}", *connection_, e.what());
```

Comment:
I would recommend setting some local boolean here and above, and then having common logic that follows for clarify, maybe also with a comment about the current state of the migration?

Reply:
Or the `status` variable could go outside of the try/catch block and catch statements can set it and the common logic of handling the error would go after the try/catch block.

Reply:
Yeah that would be fine also. Either way.

Reply:
Done